rewrite-java/src/main/java/org/openrewrite/java/cleanup/UseJavaInterface.java

```diff
@@ -0,0 +1,185 @@
+    }
+
+    @Override
+    public String getDescription() {
+        return "Replaces declarations of implemented classes in Java collection with the equivalent interface.";
+    }
+
+    @Override
+    public Set<String> getTags() {
+        return Collections.singleton("RSPEC-1319");
+    }
+
+    @Override
+    public Duration getEstimatedEffortPerOccurrence() {
+        return Duration.ofMinutes(10);
+    }
+
+    @Override
+    protected JavaVisitor<ExecutionContext> getVisitor() {
+        Map<String, String> rspecRulesReplaceTypeMap = new HashMap<>();
+        rspecRulesReplaceTypeMap.put("java.util.ArrayDeque", "java.util.Deque");
+        rspecRulesReplaceTypeMap.put("java.util.concurrent.ConcurrentLinkedDeque", "java.util.Deque");
+        // List
+        rspecRulesReplaceTypeMap.put("java.util.AbstractList", "java.util.List");
+        rspecRulesReplaceTypeMap.put("java.util.AbstractSequentialList", "java.util.List");
+        rspecRulesReplaceTypeMap.put("java.util.ArrayList", "java.util.List");
+        rspecRulesReplaceTypeMap.put("java.util.LinkedList", "java.util.List");
+        rspecRulesReplaceTypeMap.put("java.util.concurrent.CopyOnWriteArrayList", "java.util.List");
+        // Map
+        rspecRulesReplaceTypeMap.put("java.util.AbstractMap", "java.util.Map");
+        rspecRulesReplaceTypeMap.put("java.util.EnumMap", "java.util.Map");
+        rspecRulesReplaceTypeMap.put("java.util.HashMap", "java.util.Map");
+        rspecRulesReplaceTypeMap.put("java.util.Hashtable", "java.util.Map");
+        rspecRulesReplaceTypeMap.put("java.util.IdentityHashMap", "java.util.Map");
+        rspecRulesReplaceTypeMap.put("java.util.LinkedHashMap", "java.util.Map");
+        rspecRulesReplaceTypeMap.put("java.util.WeakHashMap", "java.util.Map");
+        // ConcurrentMap
+        rspecRulesReplaceTypeMap.put("java.util.concurrent.ConcurrentHashMap", "java.util.concurrent.ConcurrentMap");
+        rspecRulesReplaceTypeMap.put("java.util.concurrent.ConcurrentSkipListMap", "java.util.concurrent.ConcurrentMap");
+        // Queue
+        rspecRulesReplaceTypeMap.put("java.util.AbstractQueue", "java.util.Queue");
+        rspecRulesReplaceTypeMap.put("java.util.concurrent.ConcurrentLinkedQueue", "java.util.Queue");
+        rspecRulesReplaceTypeMap.put("java.util.concurrent.SynchronousQueue", "java.util.Queue");
+        // Set
+        rspecRulesReplaceTypeMap.put("java.util.AbstractSet", "java.util.Set");
+        rspecRulesReplaceTypeMap.put("java.util.EnumSet", "java.util.Set");
+        rspecRulesReplaceTypeMap.put("java.util.HashSet", "java.util.Set");
+        rspecRulesReplaceTypeMap.put("java.util.LinkedHashSet", "java.util.Set");
+        rspecRulesReplaceTypeMap.put("java.util.concurrent.CopyOnWriteArraySet", "java.util.Set");
+        // Sorted
+        rspecRulesReplaceTypeMap.put("java.util.TreeMap", "java.util.SortedMap");
+        rspecRulesReplaceTypeMap.put("java.util.TreeSet", "java.util.SortedSet");
```

Comment:
It's wasteful to allocate this in `getVisitor()`, which is called once per source file per cycle. Keep this information in a static field that gets initialized exactly once.